File: Javascript/49. Group Anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    let maps = new Map()
    
    for (str of strs) {
        let sorted = str.split('').sort().join('')
        
        if (maps.has(sorted)) {
            let temp = maps.get(sorted)
            temp.push(str)
            maps.set(sorted, temp)
        } else {
            maps.set(sorted, [str])
        }
    }

    return Array.from(maps.values())
};

File: Python/49. Group Anagrams.py
from collections import defaultdict

class Solution:
    def groupAnagrams(self, strs: list[str]) -> list[list[str]]:
        dicts = defaultdict(list)
        for str in strs:
            sorted_str = "".join(sorted(str))
            dicts[sorted_str].append(str)
        return list(dicts.values())
